### PR TITLE
Add CONTRIBUTING, GETTING_STARTED, and MAINTAINER_GUIDE

### LINKED ISSUE #667: Update DEVGUIDE to reflect new way of handling issues, contributing and PRs

### PR DESCRIPTION
**Pull request by TimLariviere**
Closes #667 

I've split the DEVGUIDE into several docs and updated them to reflect the new requirements, and ways we work.

Also, to prepare for v1.0, I try to introduce a new way of handling branches and releases.
(Only suggestions, I haven't done changes yet)

- master = Nightly branch with all bug fixes and features not set to a particular version, publish to MyGet (?) on each commit.
- vCurrent (e.g. `0.51`) = Current latest released version. We can add bug fixes but no breaking changes (and release a `0.51.1` for example).
- vNext (e.g. `0.52`) = Next version to release. We can add features we are sure to release next time.

vCurrent and vNext will automatically release when we create a git tag on a commit. (`git tag -a 0.52` will release the current branch as 0.52 - still need to update the release notes manually).

vNext tags should contain `-pre` to indicate it's a prerelease version.

@SergejDK What do you think of all that?

### CODE REVIEW

File: .github/GETTING_STARTED.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+## Understanding the structure
+
+There is 4 distinct libraries in this repository:
+- Fabulous
+- Fabulous.CodeGen
+- Fabulous.XamarinForms
+- Fabulous.StaticView.XamarinForms
+
+They are separated into different folders, except Fabulous which is at the root of the repository since it's its repository.
```
Comment (SergejDK) [inline .github/GETTING_STARTED.md]: I think the `since ...` part can be removed?

File: .github/GETTING_STARTED.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+To start, it has a few other folders in addition to `samples`, `src`, and `tests`.
+
+| Name  | Description                                  |
+--------|----------------------------------------------|
+| extensions | The source code of the bindings libraries of 3rd party XF libs (e.g. Maps, SkiaSharp) |
+| templates | The files used to create the `dotnet` templates (e.g. `fabulous-xf-app`) |
+| tools | The tools used to create Fabulous.XamarinForms. Most notably the bindings generator. |
+
+
+Then, since Xamarin.Forms has tens of controls with each tens of properties/events, with a new release each month or so, it is really difficult to follow and write the appropriate XF bindings manually to make it compatible with Fabulous.
```
Comment (SergejDK) [inline .github/GETTING_STARTED.md]: maybe `tons` instead of `tens`?

File: .github/GETTING_STARTED.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+Then, since Xamarin.Forms has tens of controls with each tens of properties/events, with a new release each month or so, it is really difficult to follow and write the appropriate XF bindings manually to make it compatible with Fabulous.  
+To help with that, a tool called Fabulous.XamarinForms.Generator (relying on Fabulous.CodeGen) is used to automate the generation of those bindings at build time, via a JSON file (which can be found in the Fabulous.XamarinForms project).
+
+To support this automated code generation, unlike the other libraries, Fabulous.XamarinForms compiles 2 projects into a single NuGet package: Fabulous.XamarinForms and Fabulous.XamarinForms.Core.
+
+The Core project is where you can make changes (addition, deletion, etc.). The other project is only there to bundle the Core files along with the generated XF bindings in a NuGet package.
+
+## Building
+
+Fabulous can built with a single command, making use of FAKE 5.
```
Comment (SergejDK) [inline .github/GETTING_STARTED.md]: Fabulous can be built

File: .github/MAINTAINER_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+![app size](https://user-images.githubusercontent.com/7204669/42222786-1096c20a-7ece-11e8-99d6-e1c63a6a2f30.png)
+
+Also check the app actually runs with these settings.
+
+## Continuous Integration builds
+
+This repository uses several Azure DevOps pipelines to ensure the code always builds correctly and prepare NuGet packages.
+
+Every pull request is automatically valided with a partial build on a Windows agent (compile) and a full build on a macOS agent (compile and test)
```
Comment (SergejDK) [inline .github/MAINTAINER_GUIDE.md]: validated

File: .github/MAINTAINER_GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+| t/discussion | A subject is being debated |
+| t/enhancement | This is a new feature |
+| t/bug | Something doesn't work as expected |
+| t/housekeeping | This is a repository maintenance task |
+
+#### States
+
+| Name | Description |
+|------|-------------|
+| s/unverified | The issue has not been verified yet |
```
Comment (SergejDK) [inline .github/MAINTAINER_GUIDE.md]: Will this label be automatically added for a new issue?
Should we use an automated task for this?

File: .github/GETTING_STARTED.md (hunk 2 above)
Comment (TimLariviere) [reply .github/GETTING_STARTED.md]: `dozens` is better for what I intended to say. :)

**Comment by TimLariviere** [comment]
Thanks for the proofreading!

>If we have a bugfix - should it go into all branches?

Yes!
But it follows a few rules:
- After a bug fix is sent to `vCurrent` via a PR, we will need to `git merge` `vCurrent` to `vNext` as well, followed by `git merge` of `vNext` to `master`.
- If it's a breaking change/new feature we want in the next version, we send it to `vNext` as a PR, then we `git merge` `vNext` to `master` as well.
- Any commit in `master` stays in `master`

That way, we have the correct commits in all corresponding branches.

>Should we include some automated part for placing in a project or give a label? e.g. unverified

I'm not very familiar with the automation of GitHub, but that something interesting to explore.
For `s/unverified`, it should be easy to do either through an issue template or manually once we read the issue (that's what Xamarin.Forms seems to do).

>If i am on the vNext branch and we want to prerelease, then I have to use git tag -a 0.52-pre1 or sth. like this?

Yes! Exactly.
Adding this tag will trigger the build for `vNext` and automatically push a `0.52-pre1` on NuGet.

>Prereleases with a versioning like Xamarin does?

I guess we could do that.
I'm not yet 100% clear about that, since we don't really have the need for now.

>E.g. to which branch will documentation go?

I would say `master`, since for now, https://fsprojects.github.com/Fabulous is not filterable by versions.
Though a point can be made to follow the same logic than a bug fix.
Say for example: the documentation of `v0.51` is erroneous and need to be fixed, we could commit directly to `vCurrent`.

**Comment by SergejDK** [comment]
> Yes!
But it follows a few rules:
After a bug fix is sent to vCurrent via a PR, we will need to git merge vCurrent to vNext as well, followed by git merge of vNext to master.
    If it's a breaking change/new feature we want in the next version, we send it to vNext as a PR, then we git merge vNext to master as well.
    Any commit in master stays in master
That way, we have the correct commits in all corresponding branches.
(Xamarin.Forms seems to do that as well, once a week)

Ok, got it! Very clear and easy to handle :)

> I'm not very familiar with the automation of GitHub, but that something interesting to explore.
For s/unverified, it should be easy to do either through an issue template or manually once we read the issue (that's what Xamarin.Forms seems to do).

You're right we just could use the Issue Templates -> should be enough. In GitHub you can automate like closing old issues where no discussion is made etc. Pretty cool stuff but for labeling the templates should be sufficient.

> I guess we could do that.
I'm not yet 100% clear about that, since we don't really have the need for now.

Currently I don't see the need either but for the future we should think about this.

>  E.g. to which branch will documentation go?
I would say master, since for now, https://fsprojects.github.com/Fabulous is not filterable by versions.
Though a point can be made to follow the same logic than a bug fix.
Say for example: the documentation of v0.51 is erroneous and need to be fixed, we could commit directly to vCurrent.

To handle a documentation as a bug fix is a good idea so everything is up to date and we don't have to be scared of big merge conflicts ;)

File: .github/GETTING_STARTED.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+## Testing LiveUpdate
+
+Use the CounterApp to test.  To run the equivalent of the `fabulous` CLI tool use this:
+
+    cd Fabulous.XamarinForms\samples\CounterApp\CounterApp
+    adb -d forward  tcp:9867 tcp:9867
+    dotnet run --project ..\..\..\..\src\Fabulous.Cli\Fabulous.Cli.fsproj -- --watch --send 
+
+If you want to update your (global!) install of the `fabulous-cli` tool, first bump the version number to avoid clashes, then:
```
Comment (SergejDK) [inline .github/GETTING_STARTED.md]: installation instead of install

**Comment by TimLariviere** [comment]
Thanks! I will update the documentation with what we discussed, and it will be good to merge. :)

**Comment by TimLariviere** [comment]
@SergejDK I added precisions on what we discussed. I changed a few things to be compatible with SemVer 2.0.0.

**Comment by TimLariviere** [comment]
Thanks! I merging it once build is done.

Regarding the possibility to access the documentation by branch (version number), we will need to create a custom Jekyll website (hosted on fsprojects.github.io/Fabulous still) because GitHub doesn't support that out of the box.

This will be the occasion to implement #277

**Pull request merged**